src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -32,6 +33,19 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+
+bool checkIfHiveSettingEnabled(const ContextPtr & context, const std::string & storage_type_name)
```

Comment:
It's better to store this setting in `StorageObjectStorage::QuerySettings` like other settings for engin-like storages. And set them in corresponding configurations. For example, for S3 we set them in `StorageS3Configuration::getQuerySettings`. You can look at settings `s3_throw_on_zero_files_match/hdfs_throw_on_zero_files_match/azure_throw_on_zero_files_match`